exchange/exchange.go

```diff
@@ -79,6 +81,24 @@ type bidResponseWrapper struct {
 	bidder       openrtb_ext.BidderName
 }
 
+type UUIDGenerator interface {
+	NewUUID() string
+	Enabled() bool
```

Comment:
What's the rationale behind creating this interface? Why not just add a `generateBidID` flag to the `exchange` object and do a check directly like: `if e.GenerateBidID` on L216 and if it is then directly call `uuid.NewV4().String()`?

If the purpose of this is for unit test coverage then I am wondering if we need a mock ID generator for the unit tests? Can't we just assert that the `GeneratedBidID` value is not an empty string?

Reply:
This is how it was initially, but then I modified it to interface due to this comment: https://github.com/prebid/prebid-server/pull/1772#discussion_r599699867